src/descriptor/mod.rs

```diff
@@ -511,6 +512,60 @@ impl<Pk: MiniscriptKey + ToPublicKey> Descriptor<Pk> {
     }
 }
 
+impl Descriptor<DefiniteDescriptorKey> {
+    /// Returns a plan if the provided assets are sufficient to produce a non-malleable satisfaction
+    pub fn get_plan<P>(self, provider: &P) -> Option<Plan>
```

Comment:
Does it matter that this consumes self and if `None` is returned we have no way to get the descriptor back?

Reply:
+1 on not consuming self here.

Reply:
@danielabrozzoni, I think they mean `pub fn get_plan(self, provider: &P) -> Result<Plan, Descriptor>`. That is we don't lose the descriptor in case of error .